lego/apps/notifications/tests/test_views.py

```diff
@@ -76,3 +77,148 @@ def test_change_setting_defaults(self):
+        self.authorized_user = User.objects.get(pk=9)
+        admin_group.add_user(self.authorized_user)
+        self.unauthorized_user = User.objects.get(pk=1)
+        self.authorized_user
+
+        self.unsent_announcement = Announcement.objects.get(id=5)
+        self.unsent_not_own_announcement = Announcement.objects.get(id=4)
+
+    def test_unauthorized_create(self):
+        """
+        An unauthorized user should not be able to create an Announcement
+        """
+
+        self.client.force_login(self.unauthorized_user)
+        response = self.client.post(
+            self.url, {"message": "test_message", "groups": [2], "events": [1]}
+        )
+        self.assertEquals(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_authorized_create(self):
+        """
+        An authorized user should be able to create an announcement
+        """
+
+        self.client.force_login(self.authorized_user)
+        response = self.client.post(
+            self.url,
+            {"message": "test_message", "groups": [2], "events": [1], "fromGroup": 11},
+        )
+        self.assertEquals(response.status_code, status.HTTP_201_CREATED)
+
+    def test_authorized_create_from_invalid_group(self):
+        """
+        An authorized user should not be able to create an announcement with sender as invalid group
+        """
+
+        self.client.force_login(self.authorized_user)
+        response = self.client.post(
+            self.url,
+            {"message": "test_message", "groups": [2], "events": [1], "fromGroup": 29},
+        )
+        self.assertEquals(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_authorized_create_invalid_recipient_groups(self):
+        """
+        An authorized user should not be able to create an announcement with recipient
+        as invalid group
+        """
+
+        self.client.force_login(self.authorized_user)
+        response = self.client.post(
+            self.url,
+            {"message": "test_message", "groups": [28], "events": [3], "fromGroup": 11},
+        )
+        self.assertEquals(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_authorized_patch(self):
+        """
+        It is not possible to patch an announcement
+        """
+
+        self.client.force_login(self.authorized_user)
+        response = self.client.patch(
+            self.url,
+            {
+                "id": self.unsent_announcement.id,
+                "message": "test_message",
+                "groups": [3],
+                "events": [1],
+            },
+        )
+        self.assertEquals(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_authorized_list_own(self):
+        """
+        An authorized user should be able to list announcements created by self
+        """
+
+        self.client.force_login(self.authorized_user)
+        response = self.client.get(self.url)
+        self.assertEquals(response.status_code, status.HTTP_200_OK)
```

Comment:
Check that your own announcements are returned, and that no other announcements are returned